test/e2e/common.go

```diff
@@ -425,3 +427,69 @@ func (t *Tester) GetClusterNameFromKlusterlet(klusterletName string) (string, er
 
 	return string(clusterNameByte), nil
 }
+
+func (t *Tester) OutputDebugLogs() {
+	klusterletes, err := t.OperatorClient.OperatorV1().Klusterlets().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to list klusterlets. error: %v", err)
+	}
+	for _, klusterlet := range klusterletes.Items {
+		klog.Infof("klusterlet %v : %#v \n", klusterlet.Name, klusterlet)
+	}
+
+	managedClusters, err := t.ClusterClient.ClusterV1().ManagedClusters().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("failed to list mangedClusters. error: %v", err)
+	}
+	for _, managedCluster := range managedClusters.Items {
+		klog.Infof("managedCluster %v : %#v \n", managedCluster.Name, managedCluster)
+	}
+
+	registrationPods, err := t.KubeClient.CoreV1().Pods("").List(context.Background(),
+		metav1.ListOptions{LabelSelector: "app=klusterlet-registration-agent"})
+	if err != nil {
+		klog.Errorf("failed to list registration pods. error: %v", err)
+	}
+
+	manifestWorkPods, err := t.KubeClient.CoreV1().Pods("").List(context.Background(),
+		metav1.ListOptions{LabelSelector: "app=klusterlet-manifestwork-agent"})
+	if err != nil {
+		klog.Errorf("failed to get manifestwork pods. error: %v", err)
+	}
+
+	agentPods := append(registrationPods.Items, manifestWorkPods.Items...)
+	for _, pod := range agentPods {
+		klog.Infof("klusterlet agent pod %v/%v : %#v \n", pod.Namespace, pod.Name, pod)
+		logs, err := t.PodLog(pod.Name, pod.Namespace, int64(10))
+		if err != nil {
+			klog.Errorf("failed to get pod %v/%v log. error: %v", pod.Namespace, pod.Name, err)
+			continue
+		}
+		klog.Infof("pod %v/%v logs:\n %v \n", pod.Namespace, pod.Name, logs)
+	}
+
+	manifestWorks, err := t.WorkClient.WorkV1().ManifestWorks("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
This works for now, but you may want to track specific namespaces (ie, only dump from the namespaces involved in the tests) in situations where the e2e is running against a live system in a support situation in the future.

Reply:
I add a TODO here, and will refactor this when I find a proper way to track the resources created during e2e.